src/client/editor/ConflictResolution.js

```diff
@@ -0,0 +1,97 @@
+export default class ConflictHandler {
+  constructor() {
+    this._conflicts = {
+      missing: false,
+      duplicate: false
+    };
+    this.duplicateList = {};
+  }
+
+  findDuplicates = (node, layer, index) => {
+    if (node.userData._path) {
+      node.userData._path.push(index);
+    } else {
+      node.userData._path = [0];
+    }
+
+    const name = node.name;
+    this.duplicateList[name] = name in this.duplicateList ? this.duplicateList[name] + 1 : 1;
+    if (this.duplicateList[name] > 1) {
+      this.setDuplicateStatus(true);
+    }
+
+    if (node.children) {
+      node.children.forEach((child, i) => {
+        child.userData._path = node.userData._path.slice(0);
+        this.findDuplicates(child, layer + 1, i);
+      });
+    }
+  };
+
+  setDuplicateStatus = newStatus => {
```

Comment:
If a class member is only used internally, or meant for private use, we start them with an underscore. I think that applies to a few of the methods in this class. Also applies to properties like `duplicateList`.